lesson14/main.js

```diff
@@ -71,20 +71,33 @@ const init = async () => {
   createNewList(val);
 };
 
-openModal.addEventListener("click", () => {
+const getNumbers = () => {
+  const inputNumber = document.getElementById("js-number");
+  return inputNumber.value;
+}
+
+const showModal = () => {
   modal.classList.remove("hidden");
   mask.classList.remove("hidden");
-});
+}
 
-closeModal.addEventListener("click", () => {
+const hiddenModal = () => {
   modal.classList.add("hidden");
   mask.classList.add("hidden");
+}
+
+openModal.addEventListener("click", () => {
+  showModal();
+});
+
+closeModal.addEventListener("click", () => {
+  hiddenModal();
 });
 
 requestBtn.addEventListener("click", () => {
+  console.log(getNumbers());
   init();
```

Comment:
IMO
I thought it would be better to use the init function to get the value.
> クリックした際にinput(type number)のvalueを取得して
> Promiseを実行する手前でconsole.log出力されてればいいです

This is what the problem says, but with the current code, it is displayed the moment the button is pressed.
```suggestion
requestBtn.addEventListener("click", () => {
  console.log(getNumbers());
  init(getNumbers());
```
init function

```js
const init = async (inputNumber) => {
  console.log(inputNumber); 　//Here is the console.log☺️
  createLoader();
  loading();
  const val = await fetchedData();
  createNewList(val);
};
```

Reply:
Thank you teaching me! I realized my answer was a little off from this question.

But I'm sorry I couldn't understand these code you showed me, could you teach me more? :)

- Why we separate "inputNumber" and "getNumbers" like this?

- Is it not good?

```
requestBtn.addEventListener("click", () => {
  init();
...

const init = async () => {
  console.log(getNumbers()); 
  createLoader();
...

```
`inputNumber` is include in `getNumbers` so I didn't understand why separate them.
I think I don't understand about async/await well... :(

Reply:
Thanks! Now I get it and don't worry!
I fixed them, please kindly check them :)

[2639942](https://github.com/yuka830/js-lessons/pull/16/commits/26399428f6b4dae39d99d68f61fb6cfc12616a1f)
[CodeSandbox](https://codesandbox.io/s/lesson14-fxhpe?file=/index.html)